packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -659,9 +659,8 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// {@template flutter.material.appbar.systemOverlayStyle}
   /// Specifies the style to use for the system overlays that overlap the AppBar.
   ///
-  /// If this property is null, then [SystemUiOverlayStyle.light] is used if the
-  /// overall theme is dark, [SystemUiOverlayStyle.dark] otherwise. Theme brightness
-  /// is defined by [ColorScheme.brightness] for [ThemeData.colorScheme].
+  /// If this property is null, then the appropriate [SystemUiOverlayStyle] is
+  /// calculated based on the [backgroundColor].
```

Comment:
I suppose it's OK to just say "appropriate" here, since covering the two cases for backwards compatibility would make reading the doc tedious.

Reply:
I'm not sure if I understood you correctly here, but `systemOverlayStyle` is only used for `backwardsCompatibility` disabled. Since `backwardsCompatibility` is currently still enabled by default I think it makes sense to add that to the doc. Let me know if this is better now.